### DISCUSSION TITLE
XMLSec 1.3.1 in conan center

### DISCUSSION THREAD

**Discussion by tfloodeenjr** [lsh123/xmlsec#673] [General]
Is there a possibility to get xmlsec 1.3.1 recipes for conan in conan center?

https://github.com/conan-io/conan-center-index/tree/master/recipes/xmlsec

The recipes here are out of date. I got xmlsec 1.2.37 building with OpenSSL 3.0.7 by creating my own recipe, but I am not sure I have it right. When I try to build it with OpenSSL 3.1.0, I get failures.

Is there any support for conan recipes at all?

Thanks,
-Tom

**Comment by lsh123** [comment]
I don't know anything about conan-center but XMLSec build is pretty straightforward:
```
./configure
make
make install
```

You might file a bug against conan-center, seems like there were commits only a couple months ago:

https://github.com/conan-io/conan-center-index/commits/master/recipes/xmlsec/all

**Reply by tfloodeenjr** [reply]
Yes, thanks. It seems some of those fixes help. I missed the one specific to openssl=300

                ov = Version(self.dependencies["openssl"].ref.version)
                if ov.major >= "3":
                    if Version(self.version) < "1.2.35":
                        # configure.js doesn't understand openssl=300 before xmlsec 1.2.35,
                        # For these xmlsec versions, setting 110 even for OpenSSL 3.x should be compatible
                        crypto_engines.append("openssl=110")
                    else:
                        crypto_engines.append("openssl=300")
                else:
                    crypto_engines.append(f"openssl={ov.major}{ov.minor}0")
                    
I am now able to build using xmlsec 1.2.37. At some point I will try with 1.3.1.
Regards,
-Tom
